lib/src/soloud.dart

```diff
@@ -625,6 +658,166 @@ interface class SoLoud {
     });
   }
 
+  /// Set up an audio stream.
+  ///
+  /// [maxBufferSize] the max buffer size in **bytes**. When adding audio data
+  /// using [addAudioDataStreamU8] and this values is reached, the stream will
+  /// be considered ended (likewise we called [setDataIsEnded]). This means that
+  /// when playing it, it will stop at that point (if loop is not set).
+  ///
+  /// **Note:** this parameter doesn't allocate any memory, but it just limits
+  /// the amount of data that can be added.
+  ///
+  /// [bufferingTimeNeeds] the buffering time needed in seconds. If a handle
+  /// reaches the current buffer length, it will start to buffer pausing it and
+  /// waiting until the buffer will have enough data to cover this time.
+  ///
+  /// [sampleRate] the sample rate. Usually is 22050 or 44100 (CD quality).
+  ///
+  /// [channels] enum to choose the number of channels.
+  ///
+  /// [pcmFormat] enum to choose from `f32le`, `s8`, `s16le` and `s32le`.
+  ///
+  /// [onBuffering] a callback that is called when starting to buffer
+  /// (isBuffering = true) and when the buffering is done (isBuffering = false).
+  /// It gives back the `handle` which triggered the event and the `time`
+  /// in seconds.
+  ///
+  /// Throws [SoLoudNotInitializedException] if the engine is not initialized.
+  AudioSource setBufferStream({
+    int maxBufferSize = 1024 * 1024 * 100, // 100 MB in bytes
+    double bufferingTimeNeeds = 2, // 2 seconds of data needed to un-pause
+    int sampleRate = 22050,
+    Channels channels = Channels.mono,
+    BufferPcmType pcmFormat = BufferPcmType.s16le,
+    void Function(bool isBuffering, int handle, double time)? onBuffering,
+  }) {
+    if (!isInitialized) {
+      throw const SoLoudNotInitializedException();
+    }
+
+    final ret = SoLoudController().soLoudFFI.setBufferStream(
+          maxBufferSize,
+          bufferingTimeNeeds,
+          sampleRate,
+          channels.count,
+          pcmFormat.value,
+          onBuffering,
+        );
+
+    final newSound = _addNewSound(ret.error, '', ret.soundHash.hash);
+    return newSound;
+  }
+
+  /// Add PCM audio data to the stream.
+  ///
+  /// This method can be called within an `Isolate` making possible
+  /// to create PCM data and send them to the buffer without frezing
+  /// the main thread.
```

Comment:
Interesting. Can you add a short in-line example, maybe? Or something like `See the sample code in example/blah.dart`? I don't understand how one can call a method on SoLoud from a different isolate.

Reply:
Added a short example here. Also a reference to the `example/buffer_stream/generate.dart` example (where I forgot to comment on the purpose of that :) ) which demonstrates how to generate 3 kinds of audio data using `compute`.